pages/texture/new-texture-modal.vue

```diff
@@ -350,65 +384,95 @@ export default {
 		},
 		parseJSON() {
 			try {
-				const data = JSON.parse(this.formData.importJSON);
-				this.textures = data;
+				let data = JSON.parse(this.importJSON);
+				if (!data || !data.length) data = [emptyTexture()];
+				// validate parsed data
+				const cleaned = data.map((tex) => {
+					tex.key = crypto.randomUUID();
+					tex.name ||= "";
+					tex.tags ||= [];
+					if (!tex.uses || !tex.uses.length) tex.uses = [emptyUse()];
+					tex.uses = tex.uses.map((use) => {
+						use.key = crypto.randomUUID();
+						use.edition ||= "";
+						if (!use.paths || !use.paths.length) use.paths = [emptyPath()];
+						use.paths = use.paths.map((path) => {
+							path.name ||= "";
+							path.versions ||= [];
+							path.mcmeta ||= false;
+							return path;
+						});
+						return use;
+					});
+					return tex;
+				});
+				this.textures = cleaned;
 			} catch (err) {
 				console.error(err);
 				this.$root.showSnackBar(err, "error");
 			}
 		},
+		copyData() {
+			const data = JSON.stringify(this.cleanedData, null, 2);
+			navigator.clipboard.writeText(data);
+			this.$root.showSnackBar(this.$root.lang("database.labels.copy_json_data"), "success");
+		},
+		resetModal() {
+			this.textures = [emptyTexture()];
+		},
+		closeModal() {
+			this.$emit("close");
+			this.modalOpened = false;
+		},
 		send() {
-			const data = JSON.parse(JSON.stringify(this.textures));
-			const apiData = data.map((e) => ({
-				name: e.name,
-				tags: sortTags(e.tags),
-				uses: e.uses.map((u) => ({
-					name: u.name,
-					edition: u.edition,
-					paths: u.paths.map((p) => ({
-						name: p.name,
-						versions: p.versions,
-						mcmeta: p.mcmeta || false,
-					})),
-				})),
-			}));
 			axios
```

Comment:
We're still using axios and not fetch?

Reply:
yeah the entire webapp uses axios rn, might be something to look into changing but I don't think it's worth the effort at least for now